src/general/domain/content-types/content-type.php

```diff
@@ -0,0 +1,42 @@
+<?php
+// phpcs:disable Yoast.NamingConventions.NamespaceName.TooLong -- Needed in the folder structure.
+namespace Yoast\WP\SEO\General\Domain\Content_Types;
+
+use WP_Post_Type;
+
+/**
+ * This class describes a Content Type.
+ */
+class Content_Type {
```

Comment:
This comment also applies for the Taxonomy domain object:

The point of a Domain object is that you can use it without a connection to WP. Is this case you pass the entire WP_Post_Type object to this Content_Type object. making it only a wrapper. You only need the name and label in this class. Let's just pass a name and label variable. Then you can remove the connection to WP and this Domain object just knows what it needs to know.